src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -368,11 +367,43 @@ protected virtual SamlConditions CreateConditions(SecurityTokenDescriptor tokenD
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(new Uri(tokenDescriptor.Audience)));
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
```

Comment:
This translates to:
```
 public static bool IsNullOrEmpty<T>(this IEnumerable<T> enumerable)
 {
     return enumerable == null || !enumerable.Any();
 }
```

Since we know the Audiences will never be null, just check for count > 0.

Reply:
Good point, using .Any() rather than count due to [ca1827](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1827)